lesson/outline/rtl-outline-template.md

```diff
@@ -0,0 +1,330 @@
+
+<hr>
+
+## Exercise # 3: Querying with RTL
+
+### Which course outcomes will be covered by this exercise?
+
+1. Learners will be able to perform queries with RTL helper methods.
+2. Learners will understand the difference between different types of query methods.
+
+### Narrative Summary
+
+_Check out the [content standards](http://curriculum-documentation.codecademy.com/Content-Standards/narrative/) for guidance on writing narratives for exercises._
+
+1. Explain to learners that RTL has a bunch of useful helper methods that can be used to query for DOM elements.
+2. There are two ways to perform these queries.
+- Either by the helper methods returned by the `render` function.
+```javascript
+const { getByText } = render(<div>Baz</div>)
+expect(getByText('Baz')).toBeInTheDocument()
+```
+- Or by using the `screen` object E.g.
+
+```javascript
+render(<div>Baz</div>)
+expect(screen.getByText('Baz')).toBeInTheDocument()
+```
+3. Explain that there are three types of queries (The descriptions are taken from the RTL documentation)
+    * `getBy...` :  Returns the matching node for a query, and throw a descriptive error if no elements match or if more than one match is found
+    * `queryBy...`: Returns the matching node for a query. Returns `null` if no elements match. This is useful for asserting an element that is not present.
+    * `findBy...`: Returns a Promise which resolves when an element is found which matches the given query. The promise is rejected if no element is found or if more than one element is found after a default timeout of 1000ms.
+
+    For multiple elements these are `getAllby`, `queryAllBy`, `findAllBy`
+
+4. Showcase the difference between the different query methods. E.g. the difference between `getbyText` and `querybyText`. Emphasize that instead of memorizing all these, it is best to just look them up on the [docs](https://testing-library.com/docs/queries/about/).
+5. Once learners know how to query DOM nodes with RTL, go through a few different jest matchers provided by `@testing-library/jest-dom`. Explain that these matchers are extremely helpful when looking at the state of DOM nodes in our unit tests. E.g. whether or not a button is disabled, or a checkbox is checked, etc.
+
+    Once again, it is best to look these up on the [docs](https://github.com/testing-library/jest-dom) instead of memorizing them.
+```javascript
+expect(getByTestId('button')).toBeDisabled()
+expect(getByTestId('checkbox')).toBeChecked()
+```
+6. Demonstrate how RTL gives a readable error message when a DOM component is not present or when a test has failed. The debug message can be printed with `screen.debug()`.
+
+### Checkpoints Summary
+
+_Check out the [content standards](http://curriculum-documentation.codecademy.com/Content-Standards/checkpoint/) for guidance on writing narratives for exercises._
+
+1. Extract query methods from `render`
```

Comment:
Just as an FYI, we should use `destruct` which is the term in Javascript: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

This should also be removed since we'll only teach using `screen`